code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/RegionMonitor.java

```diff
@@ -0,0 +1,85 @@
+package io.cattle.platform.servicediscovery.process;
+
+import io.cattle.platform.core.constants.CommonStatesConstants;
+import io.cattle.platform.core.model.AccountLink;
+import io.cattle.platform.core.model.Region;
+
+import static io.cattle.platform.core.model.tables.AccountLinkTable.ACCOUNT_LINK;
+import static io.cattle.platform.core.model.tables.RegionTable.REGION;
+import io.cattle.platform.db.jooq.dao.impl.AbstractJooqDao;
+import io.cattle.platform.json.JsonMapper;
+import io.cattle.platform.object.ObjectManager;
+import io.cattle.platform.object.process.ObjectProcessManager;
+import io.cattle.platform.object.util.DataAccessor;
+import io.cattle.platform.servicediscovery.service.RegionService;
+import io.cattle.platform.servicediscovery.service.impl.RegionUtil;
+import io.cattle.platform.servicediscovery.service.impl.RegionUtil.ExternalProject;
+import io.cattle.platform.servicediscovery.service.impl.RegionUtil.ExternalRegion;
+import io.cattle.platform.task.Task;
+import io.github.ibuildthecloud.gdapi.condition.Condition;
+import io.github.ibuildthecloud.gdapi.condition.ConditionType;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.inject.Inject;
+
+public class RegionMonitor extends AbstractJooqDao implements Task {
+    @Inject
+    ObjectProcessManager objectProcessManager;
+    @Inject
+    ObjectManager objectManager;
+    @Inject
+    JsonMapper jsonMapper;
+    @Inject
+    RegionService regionService;
+
+    @Override
+    public void run() {
+        List<Region> regions = objectManager.find(Region.class, REGION.REMOVED, new Condition(ConditionType.NULL));
+        if (regions.size() == 0) {
+            return;
+        }
+        List<AccountLink> accountLinks = objectManager.find(AccountLink.class, ACCOUNT_LINK.REMOVED, null, ACCOUNT_LINK.LINKED_REGION_ID, 
+            new Condition(ConditionType.NOTNULL));
+        List<String> invalidStates = Arrays.asList(CommonStatesConstants.REGISTERING, CommonStatesConstants.REMOVING, CommonStatesConstants.REMOVED);
```

Comment:
nit: you can register this as a static variable. This way we don't have to redefine it on every run